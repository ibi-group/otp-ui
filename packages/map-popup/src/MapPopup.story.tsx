import React from "react";
import { action } from "@storybook/addon-actions";
import MapPopupContents, { FocusTrapWrapper } from "./index";

export default {
  title: "Map Popup"
};

const STOP = {
  flex: false,
  id: "9526",
  lat: 45.523009,
  lon: -122.672529,
  name: "W Burnside & SW 2nd"
};

const STATION = {
  "stroke-width": 2,
  allowDropoff: true,
  allowPickup: true,
  bikesAvailable: 6,
  color: "#f00",
  id: '"hub_1580"',
  isCarStation: false,
  isFloatingBike: false,
  name: "SW Morrison at 18th",
  networks: ["BIKETOWN"],
  realTimeData: true,
  spacesAvailable: 11,
  x: -122.6896771788597,
  y: 45.5219604810172
};

const FLOATING_VEHICLE = {
  "stroke-width": 1,
  allowDropoff: false,
  allowPickup: true,
  bikesAvailable: 1,
  color: "#f00",
  id: '"bike_6861"',
  isCarStation: false,
  isFloatingBike: true,
  name: "0541 BIKETOWN",
  networks: ["BIKETOWN"],
  realTimeData: true,
  spacesAvailable: 0,
  x: -122.70486,
  y: 45.525486666666666
};

export const StopEntity = (): JSX.Element => (
  <MapPopupContents
    entity={STOP}
    setLocation={action("setLocation")}
    setViewedStop={action("setViewedStop")}
  />
);

export const StopEntityNoHandlers = (): JSX.Element => (
  <MapPopupContents entity={STOP} />
);

export const StationEntity = (): JSX.Element => (
  <MapPopupContents
    entity={STATION}
    setLocation={action("setLocation")}
    setViewedStop={action("setViewedStop")}
  />
);

export const FloatingVehicleEntity = (): JSX.Element => (
  <MapPopupContents
    entity={FLOATING_VEHICLE}
    setLocation={action("setLocation")}
    setViewedStop={action("setViewedStop")}
  />
);

export const StopEntityWithFocusTrap = (): JSX.Element => (
  <FocusTrapWrapper closePopup={() => {}} id="storybook-stop">
    <MapPopupContents
      entity={STOP}
      setLocation={action("setLocation")}
      setViewedStop={action("setViewedStop")}
    />
  </FocusTrapWrapper>
);
