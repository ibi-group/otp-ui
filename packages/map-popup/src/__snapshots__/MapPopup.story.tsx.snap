// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map Popup FloatingVehicleEntity smoke-test 1`] = `
<div class="styled__MapOverlayPopup-sc-12v7ov3-2 hLIcgt">
  <header class="styled__PopupTitle-sc-12v7ov3-4 jMeCoZ">
    Free-floating bike: 0541 BIKETOWN
  </header>
  <p class="styled__PopupRow-sc-12v7ov3-3 bGmFCM">
    <strong>
      Plan a trip:
    </strong>
    <span class="styled__FromToPickerSpan-sc-vb4790-1 hmUShj">
      <span class="styled__LocationPickerSpan-sc-vb4790-0 gwsIFy">
        <svg viewbox="0 0 512 512"
             height="0.9em"
             width="0.9em"
             aria-hidden="true"
             focusable="false"
             fill="currentColor"
             xmlns="http://www.w3.org/2000/svg"
             class="StyledIconBase-ea9ulj-0 hPhvO styled__FromIcon-sc-n5xcvc-0 jTjMOf"
        >
          <path fill="currentColor"
                d="M256 56c110.532 0 200 89.451 200 200 0 110.532-89.451 200-200 200-110.532 0-200-89.451-200-200 0-110.532 89.451-200 200-200m0-48C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 168c-44.183 0-80 35.817-80 80s35.817 80 80 80 80-35.817 80-80-35.817-80-80-80z"
          >
          </path>
        </svg>
        <button class="styled__Button-sc-vb4790-2 hNNoVB">
          From here
        </button>
      </span>
      <span class="styled__LocationPickerSpan-sc-vb4790-0 gwsIFy">
        <svg viewbox="0 0 384 512"
             height="0.9em"
             width="0.9em"
             aria-hidden="true"
             focusable="false"
             fill="currentColor"
             xmlns="http://www.w3.org/2000/svg"
             class="StyledIconBase-ea9ulj-0 hPhvO styled__ToIcon-sc-n5xcvc-2 ipFnZE"
        >
          <path fill="currentColor"
                d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
          >
          </path>
        </svg>
        <button class="styled__Button-sc-vb4790-2 hNNoVB">
          To here
        </button>
      </span>
    </span>
  </p>
</div>
`;

exports[`Map Popup StationEntity smoke-test 1`] = `
<div class="styled__MapOverlayPopup-sc-12v7ov3-2 hLIcgt">
  <header class="styled__PopupTitle-sc-12v7ov3-4 jMeCoZ">
    SW Morrison at 18th
  </header>
  <p class="styled__PopupRow-sc-12v7ov3-3 bGmFCM">
    <div>
      Available bikes: 6
    </div>
    <div>
      Available docks: 11
    </div>
  </p>
  <p class="styled__PopupRow-sc-12v7ov3-3 bGmFCM">
    <strong>
      Plan a trip:
    </strong>
    <span class="styled__FromToPickerSpan-sc-vb4790-1 hmUShj">
      <span class="styled__LocationPickerSpan-sc-vb4790-0 gwsIFy">
        <svg viewbox="0 0 512 512"
             height="0.9em"
             width="0.9em"
             aria-hidden="true"
             focusable="false"
             fill="currentColor"
             xmlns="http://www.w3.org/2000/svg"
             class="StyledIconBase-ea9ulj-0 hPhvO styled__FromIcon-sc-n5xcvc-0 jTjMOf"
        >
          <path fill="currentColor"
                d="M256 56c110.532 0 200 89.451 200 200 0 110.532-89.451 200-200 200-110.532 0-200-89.451-200-200 0-110.532 89.451-200 200-200m0-48C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 168c-44.183 0-80 35.817-80 80s35.817 80 80 80 80-35.817 80-80-35.817-80-80-80z"
          >
          </path>
        </svg>
        <button class="styled__Button-sc-vb4790-2 hNNoVB">
          From here
        </button>
      </span>
      <span class="styled__LocationPickerSpan-sc-vb4790-0 gwsIFy">
        <svg viewbox="0 0 384 512"
             height="0.9em"
             width="0.9em"
             aria-hidden="true"
             focusable="false"
             fill="currentColor"
             xmlns="http://www.w3.org/2000/svg"
             class="StyledIconBase-ea9ulj-0 hPhvO styled__ToIcon-sc-n5xcvc-2 ipFnZE"
        >
          <path fill="currentColor"
                d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
          >
          </path>
        </svg>
        <button class="styled__Button-sc-vb4790-2 hNNoVB">
          To here
        </button>
      </span>
    </span>
  </p>
</div>
`;

exports[`Map Popup StopEntity smoke-test 1`] = `
<div class="styled__MapOverlayPopup-sc-12v7ov3-2 hLIcgt">
  <header class="styled__PopupTitle-sc-12v7ov3-4 jMeCoZ">
    W Burnside &amp; SW 2nd
  </header>
  <p class="styled__PopupRow-sc-12v7ov3-3 bGmFCM">
    <strong>
      Stop ID: 9526
    </strong>
    <button class="styled__ViewStopButton-sc-12v7ov3-0 jdXQjY">
      Stop Viewer
    </button>
  </p>
  <p class="styled__PopupRow-sc-12v7ov3-3 bGmFCM">
    <strong>
      Plan a trip:
    </strong>
    <span class="styled__FromToPickerSpan-sc-vb4790-1 hmUShj">
      <span class="styled__LocationPickerSpan-sc-vb4790-0 gwsIFy">
        <svg viewbox="0 0 512 512"
             height="0.9em"
             width="0.9em"
             aria-hidden="true"
             focusable="false"
             fill="currentColor"
             xmlns="http://www.w3.org/2000/svg"
             class="StyledIconBase-ea9ulj-0 hPhvO styled__FromIcon-sc-n5xcvc-0 jTjMOf"
        >
          <path fill="currentColor"
                d="M256 56c110.532 0 200 89.451 200 200 0 110.532-89.451 200-200 200-110.532 0-200-89.451-200-200 0-110.532 89.451-200 200-200m0-48C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 168c-44.183 0-80 35.817-80 80s35.817 80 80 80 80-35.817 80-80-35.817-80-80-80z"
          >
          </path>
        </svg>
        <button class="styled__Button-sc-vb4790-2 hNNoVB">
          From here
        </button>
      </span>
      <span class="styled__LocationPickerSpan-sc-vb4790-0 gwsIFy">
        <svg viewbox="0 0 384 512"
             height="0.9em"
             width="0.9em"
             aria-hidden="true"
             focusable="false"
             fill="currentColor"
             xmlns="http://www.w3.org/2000/svg"
             class="StyledIconBase-ea9ulj-0 hPhvO styled__ToIcon-sc-n5xcvc-2 ipFnZE"
        >
          <path fill="currentColor"
                d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
          >
          </path>
        </svg>
        <button class="styled__Button-sc-vb4790-2 hNNoVB">
          To here
        </button>
      </span>
    </span>
  </p>
</div>
`;

exports[`Map Popup StopEntityNoHandlers smoke-test 1`] = `
<div class="styled__MapOverlayPopup-sc-12v7ov3-2 hLIcgt">
  <header class="styled__PopupTitle-sc-12v7ov3-4 jMeCoZ">
    W Burnside &amp; SW 2nd
  </header>
</div>
`;

exports[`Map Popup StopEntityWithFocusTrap smoke-test 1`] = `
<div id="storybook-stop-popup-focus-trap"
     role="presentation"
>
  <div class="styled__MapOverlayPopup-sc-12v7ov3-2 hLIcgt">
    <header class="styled__PopupTitle-sc-12v7ov3-4 jMeCoZ">
      W Burnside &amp; SW 2nd
    </header>
    <p class="styled__PopupRow-sc-12v7ov3-3 bGmFCM">
      <strong>
        Stop ID: 9526
      </strong>
      <button class="styled__ViewStopButton-sc-12v7ov3-0 jdXQjY">
        Stop Viewer
      </button>
    </p>
    <p class="styled__PopupRow-sc-12v7ov3-3 bGmFCM">
      <strong>
        Plan a trip:
      </strong>
      <span class="styled__FromToPickerSpan-sc-vb4790-1 hmUShj">
        <span class="styled__LocationPickerSpan-sc-vb4790-0 gwsIFy">
          <svg viewbox="0 0 512 512"
               height="0.9em"
               width="0.9em"
               aria-hidden="true"
               focusable="false"
               fill="currentColor"
               xmlns="http://www.w3.org/2000/svg"
               class="StyledIconBase-ea9ulj-0 hPhvO styled__FromIcon-sc-n5xcvc-0 jTjMOf"
          >
            <path fill="currentColor"
                  d="M256 56c110.532 0 200 89.451 200 200 0 110.532-89.451 200-200 200-110.532 0-200-89.451-200-200 0-110.532 89.451-200 200-200m0-48C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 168c-44.183 0-80 35.817-80 80s35.817 80 80 80 80-35.817 80-80-35.817-80-80-80z"
            >
            </path>
          </svg>
          <button class="styled__Button-sc-vb4790-2 hNNoVB">
            From here
          </button>
        </span>
        <span class="styled__LocationPickerSpan-sc-vb4790-0 gwsIFy">
          <svg viewbox="0 0 384 512"
               height="0.9em"
               width="0.9em"
               aria-hidden="true"
               focusable="false"
               fill="currentColor"
               xmlns="http://www.w3.org/2000/svg"
               class="StyledIconBase-ea9ulj-0 hPhvO styled__ToIcon-sc-n5xcvc-2 ipFnZE"
          >
            <path fill="currentColor"
                  d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
            >
            </path>
          </svg>
          <button class="styled__Button-sc-vb4790-2 hNNoVB">
            To here
          </button>
        </span>
      </span>
    </p>
  </div>
</div>
`;
